libraries/chain/block_log.cpp

```diff
@@ -418,8 +418,10 @@ namespace eosio { namespace chain {
       first_block_num = prune_to_num;
       block_file.flush();
 
-      fc::log_message(fc::log_context(loglevel, __FILE__, __LINE__, __func__),
-                      "blocks.log pruned to blocks ${b}-${e}", fc::mutable_variant_object()("b", first_block_num)("e", head_num));
+      const auto log_msg = fc::log_message(fc::log_context(loglevel, __FILE__, __LINE__, __func__),
+                                          "blocks.log pruned to blocks ${b}-${e}", fc::mutable_variant_object()("b", first_block_num)("e", head_num));
```

Comment:
The log_msg is created even when not used, would be better to move it into the `if`.